app/workers/backend_delete_application_worker.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class BackendDeleteApplicationWorker < ApplicationJob
+  queue_as :low
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+
+    @application = build_application(event)
+
+    [ApplicationKey, ReferrerFilter].each(&method(:delete_backend_associations))
+
+    application.delete_backend_application
+  rescue ActiveRecord::RecordNotFound => exception
+    System::ErrorReporting.report_error(exception, parameters: {event_id: event_id})
+  end
+
+  private
+
+  attr_reader :application
+
+  def build_application(event)
+    app_attrs = {id: event.application.id, application_id: event.application_id}
+    Cinstance.new(app_attrs, without_protection: true).tap do |app|
+      app.service = Service.new({id: event.service_id}, without_protection: true)
```

Comment:
This is a bit fragile, IMO. I'm sure it works now, but I confess to be worried with the layer between the service's id and `Service#backend_id`. As you know, the deletion in apisonator actually happens using `Service#backend_id`, which now reduces to `service.id.to_s`. This cannot change or it might be hard to spot the bug it would cause here.

https://github.com/3scale/porta/blob/34221313be7db9fba745d1cbfd2d9a2e97917cb0/app/models/application_key.rb#L124

https://github.com/3scale/porta/blob/34221313be7db9fba745d1cbfd2d9a2e97917cb0/app/lib/backend/model_extensions/service.rb#L12

https://github.com/3scale/porta/blob/34221313be7db9fba745d1cbfd2d9a2e97917cb0/app/models/service.rb#L272

Reply:
@guicassolato This is not meant to stay here. See this:
https://github.com/3scale/porta/pull/1830#discussion_r414215489